### DISCUSSION TITLE
How to handle (or correct?) bimodal distribution of PANDA / LIONESS edge weights

### DISCUSSION THREAD

**Discussion by LPotter21** [netZoo/netZooPy#332] [Q&A] [answered]
Good afternoon. 
I have been performing PANDA / LIONESS analysis with RNA-seq data from human tissue and have been stuck on this bimodal distribution of the edge weights seen in both our data and in the vignette data. I am using R (netZooR and pandaR), but the existing discussion questions here on netZooPy seemed in line with mine, so posting this here.

For input, I am using VST (variance stabilizing transformation) counts from DESeq2 (_C_), the "PANDA-ready motif mappings" [provided by Dr. Glass](https://sites.google.com/a/channing.harvard.edu/kimberlyglass/tools/resources) (_W_), and STRING-db physical interactions without a score threshold (_P_). I see edge weight scores split most strongly by the presence (1) or absence (0) of the edge in the motif prior.
This holds true for all sample groups in PANDA
![edgeweights](https://github.com/netZoo/netZooPy/assets/100728178/22cc8f2f-0a59-4217-b5ec-e5c71c17603c)

as well as for individual sample edge weights in LIONESS.
<img width="293" alt="edgeweights_1" src="https://github.com/netZoo/netZooPy/assets/100728178/f8da9d6b-0b79-4e2a-939f-e2e2c4a45d78">

Does this distribution of values impact the ability to determine the "real" and/or significantly different edges, in PANDA via the inverse cumulative distribution (pandaDiffEdges R command) or in LIONESS via LIMMA (linear modeling)? 
For LIONESS, I had contemplated using a cutoff (average edge weights > 0 or average edge weights > center of the dip between the mode peaks) prior to LIMMA analysis, but cutoff or not, determining which differential edges in the results were scored high enough to be "real" still isn't clear, especially if we want to identify any TF-gene interactions not present in the motif prior.

Additionally, I just discovered this [article](https://github.com/netZoo/netZooPy/blob/master/tutorials/panda/Controlling_The_Variance_Of_PANDA_Networks.ipynb) where it is clarified that it is expected for the binary nature of the motif prior to cause this strong separation in the weights. It also suggests a possible change in the analysis with the substitution of continuous values instead of binary in the motif prior. Since the posting of that article (~3 years ago), do you all know of any updated suggestions for determining input values for _W_ other than the their suggestion of motif distance to TSS?

Thank you all in advance,
Luke

**Comment by violafanfani** [comment]
Hi Luke! I think this question goes along the lines of discussion #304. I agree that we see this often in our data, where the motif is sort of leading the edge weights, but I do not have a definitive answer on how to filter them or threshold them. I think @kimberlyglass might have a better idea about thresholding and  I am also summoning @marouenbg for the discussion about non-binary priors.
Have a great day, 
Viola

**Comment by marouenbg** [comment, marked as answer]
Hi @LPotter21 ,
Thank you for your question. On continuous motif priors, we did use a continuous motif prior to mitigate some of the constraining effects of binary priors.
You can find the motif [here](https://granddb.s3.us-east-2.amazonaws.com/cells/motif/regMatCont2ProcessedEdges.csv) and you're welcome to use it but we're planning to release a deeper analysis of these motifs and how they relate to TF binding being a 'continuous' phenomenon rather than on/off. These motifs are based on the distance to TSS, but I've seen some biophysical models that compute TF affinity to the underlying promoter sequence to derive a continuous score.
Best,
Marouen

**Reply by LPotter21** [reply]
Hi Marouen, 

Thank you for the info and the data share! I'd definitely be interested in reading that analysis when it comes out, and I'll look into those biophysical model papers.

For this motif prior here, is it hg38 data? If so, was it generated in the same way as the motif prior already provided by Dr. Glass on her Google site? (CIS-BP 1.02 -> FIMO [-750,+250 TSS, q-value < 1e-4]) Just want to confirm methods in case the data works better with this motif and we want to use for publication.

**Reply by marouenbg** [reply]
Yes, hg38 but FIMO+/-1k, CIS-BP 1.94d

**Reply by LPotter21** [reply]
Awesome, thank you so much. Last questions about it then. FDR (or raw pval) cutoff? And how did you calculate the score?

**Reply by marouenbg** [reply]
It is equation 1 in methods section of https://genomebiology.biomedcentral.com/articles/10.1186/s13059-023-02877-1 q-values are integrated in the equation up to a threshold of 0.05

**Reply by LPotter21** [reply]
Ah, amazing. Thank you so much! I've been looking at that paper but had missed that piece of things.

**Reply by marouenbg** [reply]
@katehoffshutta and @lopesramos Any thoughts about a threshold for LIONESS prior to running LIMMA, see @LPotter21 's OP.
